src/structure/model.jl

```diff
@@ -222,378 +267,575 @@ end
 
 """
-Get the temperature factor on an `AbstractAtom`. Defaults to `0.0` if not read
-from the PDB file.
+Set the coordinates of an `AbstractAtom` to a `Vector{Float64}` of length 3. For
+`DisorderedAtom`s only the default atom is updated.
 """
-tempfac(atom::Atom) = atom.temp_fac
+function coords!(at::Atom, new_coords::Vector{Float64})
+    if length(new_coords) != 3
+        throw(ArgumentError("3 coordinates must be given"))
+    end
+    x!(at, new_coords[1])
+    y!(at, new_coords[2])
+    z!(at, new_coords[3])
+    return at
+end
+
+function coords!(dis_at::DisorderedAtom, coords::Vector{Float64})
+    coords!(defaultatom(dis_at), coords)
+end
+
+
+"Get the occupancy of an `AbstractAtom`. Defaults to `1.0`."
+occupancy(at::Atom) = at.occupancy
+occupancy(dis_at::DisorderedAtom) = occupancy(defaultatom(dis_at))
+
+
+"Get the temperature factor of an `AbstractAtom`. Defaults to `0.0`."
+tempfac(at::Atom) = at.temp_fac
+tempfac(dis_at::DisorderedAtom) = tempfac(defaultatom(dis_at))
+
 
 """
-Get the element of an `AbstractAtom`. Defaults to `""` if not read from the PDB
-file.
+Get the element of an `AbstractAtom`. Defaults to `\"  \"`.
+`strip` determines whether surrounding whitespace is stripped (default `true`).
 """
-element(atom::Atom) = atom.element
+function element(at::Atom; strip::Bool=true)
+    if strip
+        return Base.strip(at.element)
+    else
+        return at.element
+    end
+end
+
+element(dis_at::DisorderedAtom; strip::Bool=true) = element(defaultatom(dis_at), strip=strip)
+
 
 """
-Get the charge on an `AbstractAtom`. Defaults to `""` if not read from the PDB
-file.
+Get the charge on an `AbstractAtom`. Defaults to `\"  \"`.
+`strip` determines whether surrounding whitespace is stripped (default `true`).
 """
-charge(atom::Atom) = atom.charge
+function charge(at::Atom; strip::Bool=true)
+    if strip
+        return Base.strip(at.charge)
+    else
+        return at.charge
+    end
+end
+
+charge(dis_at::DisorderedAtom; strip::Bool=true) = charge(defaultatom(dis_at), strip=strip)
+
+
+"Get the `Residue` that an `AbstractAtom` belongs to."
+residue(at::Atom) = at.residue
+residue(dis_at::DisorderedAtom) = residue(defaultatom(dis_at))
+residue(res::AbstractResidue) = res
 
 
 """
-Determines if an `AbstractAtom` represents a hetero atom, i.e. came from a
-HETATM record, or if an `AbstractResidue` represents a hetero molecule, i.e.
-consists of HETATM records.
+Determines if an `AbstractAtom` represents a non-hetero atom, e.g. came from an
+ATOM record in a Protein Data Bank (PDB) file.
 """
-ishetero(atom::AbstractAtom) = ishetatom(atom)
+ishetatom(at::Atom) = ishetres(residue(at))
+ishetatom(dis_at::DisorderedAtom) = ishetatom(defaultatom(dis_at))
+
 
 """
 Determines if an `AbstractAtom` is a `DisorderedAtom`, i.e. if there are
 multiple locations present for an atom.
 """
-isdisorderedatom(::Atom) = false
+isdisorderedatom(at::AbstractAtom) = isa(at, DisorderedAtom)
```

Comment:
Unlike `ishetero` I think these need to stay separate, because disorder on the atom and residue levels are separate. For example a disordered residue can contain atoms in a particular residue (point mutation) that are not disordered, and disordered atoms do not need to be part of a disordered residue.